os/syscall/syscall_lookup.h

```diff
@@ -87,6 +87,10 @@ SYSCALL_LOOKUP(sem_wait,                  1, STUB_sem_wait)
 SYSCALL_LOOKUP(sem_setprotocol,           2, STUB_sem_setprotocol)
 #endif
 
+#ifdef CONFIG_BINMGR_RECOVERY
+SYSCALL_LOOKUP(sem_register,              1, STUB_sem_register)
```

Comment:
1. How about moving sem_init to kernel inside and registering it as syscall instead of sem_register?
  Because sem_init is the posix api so that it looks ok to register.
2. Could you split this as two commits? One is to move sem api to kernel api or to register syscall api, another is getting sem list in binary manager.